bm.el

```diff
@@ -555,6 +560,42 @@ before bm is loaded.")
   (interactive)
   (customize-group 'bm))
 
+(defun bm-jump ()
+  "Jump directly to any bookmark.
+Presents the bookmark list in the form of completion candidates
+in the minibuffer for quick navigation."
+ (interactive)
+ (let* ((minibuffer-history '(""))
+        (bookmarks (bm-overlays-lifo-order 'all))
+        (file-line-width (bm-find-file-line-max-width bookmarks 'all))
+        (format-string (format "%%-%ds %%-%ds" file-line-width bm-annotation-width))
+        (elements
+          (mapcar
+            (lambda (bm)
+              (with-current-buffer (overlay-buffer bm)
+                 (let ((annot (overlay-get bm 'annotation))
+                       (line-num (number-to-string
+                                   (+ 1 (count-lines (point-min) (overlay-start bm))))))
+                   (list (format "%s:%s"
+                           (propertize (buffer-name) 'face 'bm-file-name-face)
+                           (propertize line-num      'face 'bm-line-num-face))
+                         (or annot "")
+                         bm))))
+            bookmarks))
+        (candidates
+          (mapcar (lambda (elem)
+                    (list (format format-string (nth 0 elem) (nth 1 elem))
+                          (nth 2 elem)))
+                  elements))
+        decision)
+   (while (or (not decision)
+              (zerop (length decision)))
+     (setq decision
+       (completing-read
+         "Jump to bookmark: "
+         (mapcar 'car candidates) nil t)))
+   (when (setq decision (assoc decision candidates))
+     (goto-char (overlay-start (cadr decision))))))
```

Comment:
I understand that the bookmark is not available, but I don't like that jumping to a bookmark is done without using bm-goto() since there are more things happening in addition to the jump.